web-common/src/features/canvas/stores/canvas-entity.ts

```diff
@@ -62,10 +62,35 @@ export type SearchParamsStore = {
   ) => boolean;
   clearAll: () => void;
 };
+export function createComponentMap() {
```

Comment:
This is a generic util that should be in `web-common/src/lib/map-utils.ts`. We can use generics to make it have `BaseCanvasComponent` as value.